toothpick-runtime/src/main/java/toothpick/ToothPick.java

```diff
@@ -68,26 +76,25 @@ public static Scope openScope(Object name) {
    *
    * @param name the name of the scope to close.
    */
-  public static void closeScope(Object name) {
-    synchronized (ToothPick.class) {
-      Scope scope = MAP_KEY_TO_SCOPE.get(name);
-      if (scope != null) {
-        Scope parentScope = scope.getParentScope();
-        if (parentScope != null) {
-          parentScope.removeChild(scope);
-        }
-        removeScopeAndChildrenFromMap(scope);
+
+  public static synchronized void closeScope(Object name) {
```

Comment:
do you still need `synchronized` here? or only synchronized when `scope != null`